publisher-sdk/src/main/java/com/criteo/publisher/ErrorLogMessage.kt

```diff
@@ -28,20 +28,23 @@ internal object ErrorLogMessage {
   inline fun onUncaughtErrorAtPublicApi(throwable: Throwable) = LogMessage(
       Log.ERROR,
       "Internal error in ${CallerInferrer.inferCallerName()}",
-      throwable
+      throwable,
+      "onUncaughtErrorAtPublicApi"
```

Comment:
What do you think about using: 
```
object{}.javaClass.enclosingMethod.name
```
(https://stackoverflow.com/questions/48181751/get-name-of-current-function-in-kotlin)
for all the logIds? That way, if the method is renamed, the log will follow as well.

Reply:
I thought about this, but unfortunately, this won't work well:
- code is mangled by proguard: we would have method names like `a, b, c, ...`
- this is kotlin code which is kind of a sugar over JVM bytecode: internal methods are renamed to `methodName_packageName`, other renaming can occur
- in this case, the method in `inline`: you will get the name of the caller (or not if caller is Java) instead

This is too much issues I guess.

Also, I'm not sure we really want to rename the logId when method is renamed. But that's another discussion.